### PR TITLE
[#308] Upload cli binaries to S3

### PR DESCRIPTION
**Pull request by pascal-airy**

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -49,3 +49,15 @@ jobs:
...
+        uses: chrislennon/action-aws-cli@v1.1
+        env:
+          ACTIONS_ALLOW_UNSECURE_COMMANDS: 'true'
+      - name: Upload airy binary to S3
+        run: |
+          aws s3 cp bazel-bin/infrastructure/cli/airy_linux_bin s3://airy-core-binaries/`cat ./VERSION`/linux/airy
+          aws s3 cp bazel-bin/infrastructure/cli/airy_darwin_bin s3://airy-core-binaries/`cat ./VERSION`/darwin/airy
+          aws s3 cp bazel-bin/infrastructure/cli/airy_windows_bin s3://airy-core-binaries/`cat ./VERSION`/windows/airy.exe
+        env:
+          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
```
Comment (chrismatix) [inline .github/workflows/main.yml:62]: @pascal-airy I am afraid that I deleted these the other day when cleaning up the CI 😬

File: infrastructure/cli/BUILD [hunk 2]
```diff
@@ -26,3 +53,24 @@ go_test(
     embed = [":cli_lib"],
     deps = ["//infrastructure/cli/pkg/tests"],
 )
+
+genrule(
```
Comment (chrismatix) [inline infrastructure/cli/BUILD:57]: Are you sure that these rules are necessary given that they just cp 1:1? 🤔

**Review by chrismatix — DISMISSED** [review]

File: infrastructure/cli/BUILD (hunk 2 above)
Comment (pascal-airy) [reply infrastructure/cli/BUILD:57]: Yes to make the path independent from the os it is build on

File: .github/workflows/main.yml (hunk 1 above)
Comment (pascal-airy) [reply .github/workflows/main.yml:62]: I still see them. Let's see if the build fails now :)

File: infrastructure/cli/BUILD [hunk 3]
```diff
@@ -16,6 +16,33 @@ go_binary(
     visibility = ["//visibility:public"],
 )
 
+go_binary(
```
[the remark this answers is not in the document]
Comment (pascal-airy) [reply infrastructure/cli/BUILD]: I renamed it

File: infrastructure/cli/BUILD (hunk 2 above)
Comment (chrismatix) [reply infrastructure/cli/BUILD]: Ok, so up to you, but you can also use `$(bazel info bazel-genfiles)` to obtain the path to the out folder without having to hardcode the os specific path

**Review by lucapette — DISMISSED** [review]

**Review by ljupcovangelski — DISMISSED** [review]

File: infrastructure/cli/BUILD (hunk 2 above)
Comment (pascal-airy) [reply infrastructure/cli/BUILD]: Thanks, Chris I will keep it in mind 👍

**Review by lucapette — APPROVED** [review]

**Pull request merged**